### ISSUE TITLE
Encapsulate dependent projects under a dedicated node

### ISSUE THREAD

**Issue by tmat** [KirillOsenkov/MSBuildStructuredLog#358]
Currently, the nodes that represent dependencies of a given project are listed directly under the project's node as shown below. The main target of the project is then listed after all these dependencies. This list can get very long making it harder to navigate around. 

![image](https://user-images.githubusercontent.com/41759/84452742-87bab780-ac0b-11ea-87d1-d3529eca1c01.png)
...
![image](https://user-images.githubusercontent.com/41759/84453543-ed5b7380-ac0c-11ea-8e7f-6a85e16cce3b.png)

**Proposal**

Group all dependencies under "Dependencies" node similarly to "Properties".

```
> Project "InteractiveHost.UnitTests.csproj" (default targets):
   > Properties
   > Dependencies
     > Project "Microsoft.CodeAnalysis.csproj" (GetTargetFrameworks targets(s)):
     ... 
     > Project "RepositoryValidation.proj" (Validate targets(s)):
   > Target Build
```

Additional tweak: add another level of grouping by project:

```
> Project "InteractiveHost.UnitTests.csproj" (default targets):
   > Properties
   > Dependencies
     > Project "Microsoft.CodeAnalysis.csproj" targets
       > default targets
       > GetTargetFrameworks target
       > GetNativeManifest target
       > GetCopyToOutputDirectoryItems target
     ... 
     > Project "RepositoryValidation.proj"
       > Validate target
   > Target Build
```

**Comment by tmat** [comment]
@KirillOsenkov

**Comment by KirillOsenkov** [comment]
This actually seems like a bug in MSBuild:

https://github.com/microsoft/msbuild/issues/5422

These projects are supposed to be nested under the MSBuild task that spawned them (ResolveProjectReferences -> MSBuild in this case):

![image](https://user-images.githubusercontent.com/679326/84455305-dbc89a80-ac11-11ea-85d0-cd904c967377.png)

**Comment by tmat** [comment]
Thanks for looking into it!

**Comment by KirillOsenkov** [comment]
This should have been fixed as of 16.9.
